modules/augment/src/augmenter.cpp

```diff
@@ -0,0 +1,169 @@
+            }
+
+        }
+
+        dstImgs.create(augmentedImg.size(), augmentedImg.type(), i, true);
+        Mat dstImg = dstImgs.getMat(i);
+        augmentedImg.copyTo(dstImg);
+
+        dstMasks.create(augmentedMask.size(), augmentedMask.type(), i, true);
+        Mat dstMask = dstMasks.getMat(i);
+        augmentedMask.copyTo(dstMask);
+
+    }
+}
+
+
+void Augmenter::applyImagesWithPoints(InputArrayOfArrays imgs,
+    InputArrayOfArrays points,
+    OutputArrayOfArrays dstImgs,
+    OutputArrayOfArrays dstPoints)
+{
+    dstImgs.create(imgs.cols(), 1, 0, -1, true);
+    dstPoints.create(points.cols(), 1, 0, -1, true);
+    RNG rng;
+
+    for (int i = 0; i < imgs.cols(); i++)
+    {
+        Mat originalImg = imgs.getMat(i);
+        Mat originalPoints = points.getMat(i);
+        Mat augmentedImg = originalImg.clone();
+        Mat augmentedPoints = originalPoints.clone();
+
+        for (unsigned int j = 0; j < transformations.size(); j++)
+        {
+            float prob = rng.uniform(0.f, 1.f);
+
+            if (prob <= probs[j])
+            {
+                transformations[j]->init(augmentedImg);
+                transformations[j]->image(augmentedImg, augmentedImg);
+                transformations[j]->points(augmentedPoints, augmentedPoints);
+            }
+
+        }
+
+        dstImgs.create(augmentedImg.size(), augmentedImg.type(), i, true);
+        Mat dstImg = dstImgs.getMat(i);
+        augmentedImg.copyTo(dstImg);
+
+        dstPoints.create(augmentedPoints.size(), augmentedPoints.type(), i, true);
+        Mat dstImgPoints = dstPoints.getMat(i);
+        augmentedPoints.copyTo(dstImgPoints);
+
+    }
+}
+
+
+void Augmenter::applyImagesWithRectangles(InputArrayOfArrays imgs,
+    InputArrayOfArrays rects,
+    OutputArrayOfArrays dstImgs,
+    OutputArrayOfArrays dstRects)
+{
+    dstImgs.create(imgs.cols(), 1, 0, -1, true);
+    dstRects.create(rects.cols(), 1, 0, -1, true);
+    RNG rng;
+
+    for (int i = 0; i < imgs.cols(); i++)
+    {
+        Mat originalImg = imgs.getMat(i);
+        Mat originalRects = rects.getMat(i);
+        Mat augmentedImg = originalImg.clone();
+        Mat augmentedRects = originalRects.clone();
+
+        for (unsigned int j = 0; j < transformations.size(); j++)
+        {
+            float prob = rng.uniform(0.f, 1.f);
+
+            if (prob <= probs[j])
+            {
+                transformations[j]->init(augmentedImg);
+                transformations[j]->image(augmentedImg, augmentedImg);
+                transformations[j]->rectangles(augmentedRects, augmentedRects);
+            }
+
+        }
+
+        dstImgs.create(augmentedImg.size(), augmentedImg.type(), i, true);
+        Mat dstImg = dstImgs.getMat(i);
+        augmentedImg.copyTo(dstImg);
+
+        dstRects.create(augmentedRects.size(), augmentedRects.type(), i, true);
+        Mat dstImgRects = dstRects.getMat(i);
+        augmentedRects.copyTo(dstImgRects);
+
+    }
+}
```

Comment:
Please reduce code duplication: every method looks very similar to each other.

Reply:
that;s because the only difference is the type of ground truth data